File: 2-year/java/fiblioteca/src/repositories/implementations/StudentsRepository.java
package repositories.implementations;

import entities.Student;
import repositories.interfaces.IStudentsRepository;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

public class StudentsRepository implements IStudentsRepository {
    private Repository repository;

    public StudentsRepository() {
        this.repository = Repository.getInstance();
    }

    @Override
    public void create(Student student) {
        try {
            PreparedStatement statement = this.repository.getConnection()
                .prepareStatement(
                    "INSERT INTO Students ("
                        + "ra,"
                        + "name "
                        + ") VALUES (?,?);"
                );

            statement.setInt(0, student.getRa());
            statement.setString(1, student.getName());

            statement.execute();
        } catch (SQLException err) {
            err.printStackTrace();
        }
    }

    @Override
    public ArrayList<Student> findAll() {
        ArrayList<Student> result = new ArrayList<Student>();

        try {
            ResultSet lines = this.repository
                .getConnection()
                .prepareStatement("SELECT * FROM Students")
                .executeQuery();

            while (lines.next()) {
                int ra = lines.getInt("ra");
                String name = lines.getString("name");

                Student student = new Student(ra, name);

                result.add(student);
            }
        } catch (SQLException err) {
            err.printStackTrace();
        } finally {
            return result;
        }
    }

    @Override
    public Student findByRa(int ra) {
        try {
            ResultSet lines = this.repository
                .getConnection()
                .prepareStatement("SELECT * FROM Students WHERE ra = " + ra)
                .executeQuery();

            lines.next();

            String name = lines.getString("name");

            return new Student(ra, name);
        } catch (SQLException err) {
            err.printStackTrace();
        } finally {
            return null;
        }
    }
}

File: 1-year/python/school-system/src/repositories/students_repository.py
repository = []


def create(data):
    repository.append(data)


def findById(id):
    result = None

    for student in repository:
        if (student.id == id):
            result = student
            break

    return result


def findByRa(ra):
    result = None

    for student in repository:
        if (student.ra == ra):
            result = student
            break

    return result


def findByCpf(cpf):
    result = None

    for student in repository:
        if (student.cpf == cpf):
            result = student
            break

    return result


def findAll():
    return repository


def updateByRa(ra, data):
    global repository

    filtered = list(filter(lambda student: student.ra != ra, repository))
    filtered.append(data)

    repository = filtered


def delById(id):
    global repository

    filtered = list(filter(lambda student: student.id != id, repository))

    repository = filtered
